src/parser/data-purge-module.ts

```diff
@@ -33,6 +33,8 @@ export class DataPurgeModule extends BaseModule {
           .replace(/^\/.+/g, "")
           // Remove HTML comments
           .replace(/<!--[\s\S]*?-->/g, "")
+          // Remove the footnotes
+          .replace(/^###### .*?\[\^\d+\^][\s\S]*$/gm, "")
```

Comment:
This approach seems wrong. You should check revision history and remove any content added by bots or other users.

Reply:
The problem with that is the bot comments multiple times and your comment also get tangled with the bot edits. If you look at the referenced PR, here is the last edit by yourself
<img width="647" alt="image" src="https://github.com/user-attachments/assets/95fa07a1-a9a5-4326-ab37-79ef745b8647">

For reference, you can check with the following GQL query:
```gql
query CommentHistory($commentId: ID!) {
  node(id: $commentId) {
    ... on Issue {
      userContentEdits(first: 100) {
        nodes {
          editedAt
          editor {
            login
          }
          diff
          createdAt
          updatedAt
        }
      }
    }
  }
}
```

So in the end I would credit you these links because they appear in your revision history. I am not sure how to fix this.

Reply:
@sshivaditya any ideas on how to improve the bot edit implementation? 

Otherwise we might be able to annotate the inserted links in a specific way to at least handle this situation

Reply:
I checked other comments and some are mixed, some are not. Maybe you changed a tab by accident when editing or something. In https://github.com/ubiquity-os-marketplace/text-conversation-rewards/issues/143 for example it is ok. So either we keep the regex to cleanup because it is more accurate, either we use the diffs with the risk of rewarding links that should not be?

Reply:
I would try implementing the diff approach and if it is too difficult to get it to work reliably then I would go with your current implementation.

Reply:
Reliably depends mostly on GitHub API I suppose, sometimes I must say it looks weird because the "diff" is just the comment you posted not a diff formatted content. Making a separate spec to explore this.

Reply:
Then it makes more sense to keep it generic for now and remove the footnotes entirely. I will try to implement an effective diff in a separate PR.